src/main/java/CommandGenerator.java

```diff
@@ -0,0 +1,19 @@
+import java.util.Optional;
+
+public class CommandGenerator{
+    public static Optional<Command> generate(String input) {
+        switch (Parser.analyzeInput(input)) {
```

Comment:
Just a nitpick, but might be better to split lines into more readable chunks. Noticed the same across the files as well :)
Something like: 
```suggestion
        Parser parsedInput = Parser.analyzeInput(input);
        switch (parsedInput) {
```